src/widgetastic_patternfly/__init__.py

```diff
@@ -2351,3 +2351,111 @@ def item_select(self, item, close=True):
         finally:
             if close:
                 self.close()
+
+
+class ItemsList(View):
+    """Basic list-view handling class:
+    https://www.patternfly.org/pattern-library/content-views/list-view/
+    Most functionality is meant to mimic widgetastic's Table class.
+    This class is meant to work with an ItemClass that is defined elsewhere. The ItemClass
+    is a ParametrizedView that represents a single item in the list-view
+    (similar to TableRow and Table). The parameter for the ItemClass is it's index/position in
+    the list-view.
+
+    In practice, the ItemClass will be defined within a nested view e.g. in a view class we could
+    have:
+    .. code-block:: python
+        # define a view
+        class MyView(BaseLoggedInPage):
+            # define the list-view widget that's on this page
+            @View.nested
+            class item_list(ItemsList):
+                # define the item_class that the list-view uses
+                item_class = ItemClass
+            # define whatever else this page has on it ...
+
+    For an example: integration_tests/cfme/control/explorer/alerts.py::MonitorAlertsAllView
+
+    Usage is as follows assuming the list is instantiated as ``view.item_list``:
+
+    .. code-block:: python
+        # Access item by position
+        view.item_list[0].next() # => gives first item in list-view
```

Comment:
indexed and next call to pull from generator, why is that? I'd expect item_list[0] to give me the first row?

Reply:
This is because of the fact that the method `items()` is a generator,`__getitem__` is designed to be used with a general `item_filter`, not necessarily just an integer. I agree that if `item_filter` happens to be an integer, we should definitely return that row. This behavior can be included in `__getitem__`